File: nvpl_blas/f77/zscal.f
!******************************************************************************
! Content:
!     This example demonstrates use of API as below:
!     ZSCAL 
!
!******************************************************************************
      program   ZSCAL_MAIN
*
      implicit none

      include "nvpl_blas_f77_blas.fi"

*       Intrinsic Functions
      intrinsic  abs

      integer         n, incx, i
      parameter       (n=5, incx=1)
      double complex  alpha
      parameter       (alpha=1.0)
      integer         xsize
      parameter       (xsize=1+(n-1)*abs(incx))
      double complex  x(xsize)
      data            (x(i), i=1,xsize)/1.0,-3.7,4.3,0.0,5.0/

*       External Subroutines
      external        print_zvector 
*
*      Executable Statements
*
      print*
      print 99
*
*       Print input data
      print*
      print 100, n, incx, alpha
      call print_zvector(x,n,incx,'X')
*
*      Call ZSCAL subroutine
      call ZSCAL(n,alpha,x,incx)
*
      print*
      print 101
      call print_zvector(x,n,incx,'X')

      stop
 99   format('Example: ZSCAL for multiplying the vector x by the'
     &       ' scalar')
 100  format('#### args: n=',i1,', incx=',i1,', '
     &       'alpha=(',f3.1,', ',f3.1,')')
 101  format('The result of ZSCAL: ')
      end

File: nvpl_blas/c/zscal.c
/******************************************************************************
 * Content: 
 *     This example demonstrates use of API as below:
 *     cblas_zscal
 *
 ******************************************************************************/
#include "example_helper.h"

int main() {
    nvpl_int_t N = 5;
    nvpl_dcomplex_t * X;
    nvpl_dcomplex_t alpha = {1.0f, 2.0f};
    nvpl_int_t incX = 1;

    printf("\nExample: cblas_zscal for multiplying the vector x by the scalar\n");
    printf("#### args: n=%" PRId64 ", incx=%" PRId64 ", alpha=(%g, %g)\n", (int64_t)N,
            (int64_t)incX, alpha.real, alpha.imag);

    nvpl_int_t len_x = 1 + (N - 1) * labs(incX);
    // allocate memory
    X = (nvpl_dcomplex_t *)malloc(len_x * sizeof(nvpl_dcomplex_t));

    // fill data
    fill_zvector(X, N, incX);

    // print input data
    print_zvector(X, N, incX, "X");

    // call cblas_zscal
    cblas_zscal(N, &alpha, X, incX);

    // print result
    printf("\nThe result of cblas_zscal:\n");
    print_zvector(X, N, incX, "X");

    // release memory
    free(X);
    return 0;
}
